### DISCUSSION TITLE
Start Stellarium on extended desktop

### DISCUSSION THREAD

**Discussion by mkoch44** [Stellarium/stellarium#2494] [General]
Is it possible to start Stellarium automatically
-- full-screen on the extended 3840x2160 screen, if an external 4K beamer is connected
-- full-screen on the default 1920x1080 screen, if the beamer is not connected?

Or do I need two different configuration files?

Thanks,
Michael

**Comment by mkoch44** [comment]
P.S. The desktop layout is 1920x1080 at the left side and 3820x2160 at the right side. The user interface, from where I start Stellarium, is on the left side. The right side is a 4K beamer with a fisheye lens in a planetarium dome.
I want to use the "extended desktop" mode, because in "duplicate screen" mode all the icons on my desktop would become visible in the dome.

**Comment by gzotti** [comment]
I don't think this auto configuration is possible, but you have the hardware to try it out. 
Configure setup A, then store settings, exit Stellarium. Launch Stellarium, see if it is on the second screen.
Disconnect 4k screen, launch Stellarium, and see it fullscreen on the FullHD screen. If you store settings now, I think it may store screen1. But if you don't store settings, maybe it will remember to display on the 4k screen.

**Reply by mkoch44** [reply]
Thank you very much for your answer. It does indeed work as you described. I did test with Windows 10:
Start Stellarium. If it's in fullscreen mode, press F11 to make it a window.
Move this window to the extended desktop (4K beamer), then press F11 to make it fullscreen.
Press F2 to open the configuration window. Click on "Save Settings" (which is in the middle of the screen). It's not possible to open the configuration window with the mouse, because the lower left corner of the screen isn't visible in the dome).
That's all and Stellarium does now behave as follows:
If the beamer is not connected, it will start fullscreen on the default FHD screen.
If the 4K beamer is connected, it will start on the extended desktop.
If the beamer is disconnected while Stellarium is running, it does automatically move fullscreen to the default FHD screen.
If the beamer is connected while Stellarium is running, it does not automatically move to the extended desktop. You have to restart Stellarium.
Suggestion for improvement: Give "Save settings" a keyboard shortcut.
